semeio/jobs/overburden_timeshift/ots.py

```diff
@@ -100,18 +102,35 @@ def ots_run(parameter_file):
         )
         line = "{}, {}, {}" + ", {}" * num_pairs + "\n"
         with open(parms.vintages_export_file, "w") as f:
-            for point in range(len(surface_horizon)):
-                xy = surface_horizon.getXY(point)
-                ts = []
-                for iv in range(len(vintage_pairs.ts)):
-                    ts.append(tshift_ts[iv][point])
-                for iv in range(len(vintage_pairs.ts_simple)):
-                    ts.append(tshift_ts_simple[iv][point])
-                for iv in range(len(vintage_pairs.dpv)):
-                    ts.append(tshift_dpv[iv][point])
-                for iv in range(len(vintage_pairs.ts_rporv)):
-                    ts.append(tshift_ts_rporv[iv][point])
-                f.write(line.format(xy[0], xy[1], surface_horizon[point], *ts))
+
+            point = 0
+            for x_index in range(1, surface_horizon.get_nx() + 1):
```

Comment:
We tend to do a lot of these double loops, which is a very natural thing as we are working with grids very often (or even triple for loops, when going through 3 dimensions).

I've always disliked that it is necessary to add more indents than a single one.. 

What do you think about:

`for i, j in itertools.product(range(1, nx+1), range(1, ny+1)):`

A bit annoying that it isn't 0-indexed, but alright..

If you need the `point` as well:

`for point, (i, j) in enumerate(itertools.product(range(1, nx+1), range(1, ny+1))):`

Reply:
Yes, will probably do something like this, or perhaps suggest a patch upstream? I think it makes sense for a surface to be iterable, and this is at least what I would expect to get if I iterated over it (it is not iterable at the moment).

Reply:
My first goal was just to replace it, then clean up the implementation ;)